libraries/chain_kv/include/b1/session/rocks_session.hpp

```diff
@@ -74,11 +80,18 @@ class session<rocksdb_t> {
    session()               = default;
    session(const session&) = default;
    session(session&&)      = default;
+
+   /// \brief Constructor
+   /// \param db A pointer to the RocksDB db type instance.
+   /// \param max_iterators This type will cache up to max_iterators RocksDB iterator instances.
    session(std::shared_ptr<rocksdb::DB> db, size_t max_iterators);
 
    session& operator=(const session&) = default;
    session& operator=(session&&) = default;
 
+   std::unordered_set<shared_bytes> updated_keys() const;
```

Comment:
I think putting definitions within the class declaration cause a lot of clutter and makes it harder to see what the definition of the type is. Technically I could just move the definitions into a source file so if/when I get around to that it would be much simpler to do having this split between definition and declaration.